doc/api/kernel_api.rst

```diff
@@ -238,3 +238,17 @@ of variable-size data items.
    :project: Zephyr
    :content-only:
 
+Memory Domain
+*************
+
+A memory domain would contain some number of memory partitions.
```

Comment:
Better to speak in the present tense:

     A memory domain contains some number of memory partitions. Threads can
     specify the range and attribute (access permission) for memory partitions in a
     memory domain. Threads in the same memory domain have the same access
     permissions to the memory partitions belong to the memory domain.
    (See :ref:`memory_domain`.)